### PR TITLE
[DoctrineBridge] Fix deprecation warning with ORM 3 when guessing field lengths

### PR DESCRIPTION
**Pull request by eltharin**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

 [DoctrineBridge] prevent deprecated message #54255  for symfony 5.4

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php [hunk 1]
```diff
@@ -141,8 +141,10 @@ public function guessMaxLength(string $class, string $property)
         if ($ret && isset($ret[0]->fieldMappings[$property]) && !$ret[0]->hasAssociation($property)) {
             $mapping = $ret[0]->getFieldMapping($property);
 
-            if (isset($mapping['length'])) {
-                return new ValueGuess($mapping['length'], Guess::HIGH_CONFIDENCE);
+            $length = \is_array($mapping) ? $mapping['length'] : $mapping->length;
```
Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]: The previous logic allowed the `length` key not to be set. Your code would trigger a warning in that case.

Comment (derrabus) [reply src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]: What if we tested for the actual class instead?

```php
$length = $mapping instanceof FieldMapping ? $mapping->length : ($mapping['length'] ?? null);
```

File: src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php [hunk 2]
```diff
@@ -141,8 +141,10 @@ public function guessMaxLength(string $class, string $property)
...
+
+            if (isset($length)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]:
```suggestion
            if (null !== $length) {
```

File: src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php [hunk 3]
```diff
@@ -141,8 +142,10 @@ public function guessMaxLength(string $class, string $property)
         if ($ret && isset($ret[0]->fieldMappings[$property]) && !$ret[0]->hasAssociation($property)) {
             $mapping = $ret[0]->getFieldMapping($property);
 
-            if (isset($mapping['length'])) {
-                return new ValueGuess($mapping['length'], Guess::HIGH_CONFIDENCE);
+            $length = $mapping instanceof FieldMapping ? $mapping->length : ($mapping['length'] ?? null);
+
+            if (null == $length) {
```
Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]:
```suggestion
            if (null !== $length) {
```

**Review by derrabus — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @eltharin.

**Pull request merged**